### PR TITLE
Modify comments to include a name field.

### PR DESCRIPTION
**Pull request by ganewman**
Allow user to include a name when making a comment, which is then displayed. Also create a new "Comment" class to facilitate this as well as future communication with database.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -28,13 +29,10 @@
...
 public class DataServlet extends HttpServlet {
-  private ArrayList<String> comments;
+  private ArrayList<Comment> comments;
 
   public DataServlet() {
-    comments = new ArrayList<>(3);
-    comments.add("Hello!");
-    comments.add("Nice to meet you!");
-    comments.add("Have a nice day!");
+    comments = new ArrayList<>(5);
```
Comment (suparnabarua) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:35]: Don't set the capacity of the array list unless you're immediately adding contents to it. Now comments is being grown dynamically in doPost

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -45,19 +43,21 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     
     Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("text", userComment);
-    commentEntity.setProperty("time", timestamp);
-    
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("content", content);
+    commentEntity.setProperty("time", time);
+
     // TODO remove once code is able to read from database
```
Comment (suparnabarua) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: Is this TODO still relevant?

Comment (ganewman) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:59]: Unfortunately yes. My next pull request should add the code to actually read from the database.

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -68,13 +68,24 @@ async function getCommentsFromServlet() {
...
+
+  const nameElement = document.createElement('h4');
+  nameElement.innerText = commentObj.name;
+  commentDiv.appendChild(nameElement);
+
+  const contentElement = document.createElement('p');
+  contentElement.innerText = commentObj.content;
+  commentDiv.appendChild(contentElement);
+
+  // TODO: do something useful with the time information
```
Comment (kellyktan) [inline portfolio/src/main/webapp/script.js:88]: You could maybe display the time when hovering over the comment? https://www.w3schools.com/css/css_tooltip.asp

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  private final String name;
+  private final String content;
+  private final long time;
+
+  public Comment(String name, String content, long time) {
+    this.name = name;
+    this.content = content;
+    this.time = time;
+  }
+}
```
Comment (kellyktan) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:28]: Add newline at end of file.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (ganewman) [reply portfolio/src/main/webapp/script.js:88]: That could be cool! I might add that later.

**Pull request merged**